src/coord/src/catalog/builtin.rs

```diff
@@ -1237,6 +1236,30 @@ WHERE
     mz_dataflow_names.worker = mz_dataflow_operator_addresses.worker",
 };
 
+pub const MZ_CLUSTER_REPLICAS: BuiltinView = BuiltinView {
+    name: "mz_cluster_replicas",
+    schema: MZ_CATALOG_SCHEMA,
+    sql: "CREATE VIEW mz_catalog.mz_cluster_replicas AS
+WITH counts AS (
+    SELECT
+        replica_id,
+        sum(CASE WHEN status = 'ready' THEN 1 else 0 END) AS ready,
+        sum(CASE WHEN status = 'ready' THEN 0 else 1 END) AS not_ready
+    FROM mz_catalog.mz_cluster_replicas_status
+    GROUP BY replica_id
+)
+SELECT
+    mz_cluster_replicas_base.*,
+    CASE
+        WHEN counts.ready > 0 AND counts.not_ready = 0
+            THEN 'healthy'
+            ELSE 'unhealthy'
+        END AS status
+FROM mz_catalog.mz_cluster_replicas_base
+LEFT OUTER JOIN counts
+    ON mz_cluster_replicas_base.id = counts.replica_id",
```

Comment:
This view extends the `mz_cluster_replicas_base` table by a column `status` that contains "healthy" for a replica if all the replica's processes are ready in `mz_cluster_replicas_status` *and* there is at least one process, and "unhealthy" otherwise.

Let me know if you can think of a more succinct or efficient way to write this!

Reply:
Seems pretty clear/efficient to me